compiler/rustc_typeck/src/collect.rs

```diff
@@ -2778,6 +2778,13 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
         }
     }
 
+    // The panic_no_unwind function called by TerminatorKind::Abort will never
+    // unwind. If the panic handler that it invokes unwind then it will simply
+    // call the panic handler again.
+    if Some(id) == tcx.lang_items().panic_no_unwind() {
+        codegen_fn_attrs.flags |= CodegenFnAttrFlags::NEVER_UNWIND;
```

Comment:
Could we use `#[rustc_allocator_nounwind]` instead? There's nothing allocator-specific for that attribute even though it has allocator in its name.

Reply:
If we did this, I would suggest renaming the attribute.

Reply:
I agree that this should be an attribute (or possibly a `extern "Rust-nounwind"` ABI?). I've been away this past week and haven't had time to implement this yet, but it is on my TODO list.